addons/binding/org.openhab.binding.miele/src/main/java/org/openhab/binding/miele/handler/HobChannelSelector.java

```diff
@@ -35,22 +35,55 @@
     STATE("state", "state", StringType.class, false),
     PLATES("plateNumbers", "plates", DecimalType.class, true),
     PLATE1_POWER("plate1PowerStep", "plate1power", DecimalType.class, false),
-    PLATE1_HEAT("plate1RemainingHeat", "plate1heat", DecimalType.class, false),
+    PLATE1_HEAT("plate1RemainingHeat", "plate1heat", DecimalType.class, false) {
+        @Override
+        public State getState(String s, DeviceMetaData dmd) {
+            // If there is remaining heat, the device metadata contains some informative string which can not be
+            // converted into a DecimalType. We therefore ignore the metadata and return the device property value as a
+            // State
+            return getState(s);
+        }
+    },
     PLATE1_TIME("plate1RemainingTime", "plate1time", StringType.class, false),
     PLATE2_POWER("plate2PowerStep", "plate2power", DecimalType.class, false),
-    PLATE2_HEAT("plate2RemainingHeat", "plate2heat", DecimalType.class, false),
+    PLATE2_HEAT("plate2RemainingHeat", "plate2heat", DecimalType.class, false) {
+        @Override
+        public State getState(String s, DeviceMetaData dmd) {
+            return getState(s);
+        }
+    },
     PLATE2_TIME("plate2RemainingTime", "plate2time", StringType.class, false),
     PLATE3_POWER("plate3PowerStep", "plate3power", DecimalType.class, false),
-    PLATE3_HEAT("plate3RemainingHeat", "plate3heat", DecimalType.class, false),
+    PLATE3_HEAT("plate3RemainingHeat", "plate3heat", DecimalType.class, false) {
+        @Override
+        public State getState(String s, DeviceMetaData dmd) {
+            return getState(s);
+        }
+    },
     PLATE3_TIME("plate3RemainingTime", "plate3time", StringType.class, false),
     PLATE4_POWER("plate4PowerStep", "plate4power", DecimalType.class, false),
-    PLATE4_HEAT("plate4RemainingHeat", "plate4heat", DecimalType.class, false),
+    PLATE4_HEAT("plate4RemainingHeat", "plate4heat", DecimalType.class, false) {
+        @Override
+        public State getState(String s, DeviceMetaData dmd) {
+            return getState(s);
+        }
+    },
     PLATE4_TIME("plate4RemainingTime", "plate4time", StringType.class, false),
     PLATE5_POWER("plate5PowerStep", "plate5power", DecimalType.class, false),
-    PLATE5_HEAT("plate5RemainingHeat", "plate5heat", DecimalType.class, false),
+    PLATE5_HEAT("plate5RemainingHeat", "plate5heat", DecimalType.class, false) {
```

Comment:
I think that there better options to resolve this with less duplication

Reply:
@martinvw how do you mean?

Reply:
This whole enum actually looks like a matrix to me.

So we have 6 instances of PLATE_HEAT, PLAT_TIME & PLATE_POWER

So maybe objects would be more suitable, but if we stay on this enum, we could consider adding a additional boolean to state that `deviceMetaData` should be ignored. You could also offer some really distinct constructors (a different subject for example by adding a marker type to it).

Hereby some abstract idea to also simplify the existing constructor calls instead of making them more complex by just adding the extra boolean

```
PLATE5_HEAT(RemainingHeat.REMAINING_HEAT, 5),
PLATE5_POWER(PlatePower.POWER, 5),
;

private enum RemainingHeat { REMAINING_HEAT }
private enum PlatePower { POWER }

     HobChannelSelector(RemainingHeat selector, int nr) {
         this("plate"+nr+"RemainingHeat", "plate"+nr+"heat", DecimalType.class, false, false);
     }

     private HobChannelSelector(String propertyID, String channelID, Class<? extends Type> typeClass, boolean isProperty, boolean ignoreDeviceMetaData) {
         this.mieleID = propertyID;
         this.channelID = channelID;
         this.typeClass = typeClass;
         this.isProperty = isProperty;
         this.ignoreDeviceMetaData = ignoreDeviceMetaData;
     }
```

Reply:
@martinvw you are too much of a code purist. Fact is that the Miele protocol is not yet fully reverse-engineered, and I prefer to get it working/complete instead of doing these kinds of code optimisations. Besides that, Miele itself is evolving their protocol, and it could very well be the case this binding will be thrown out of the window (In fact, I do know that this will be the case but that is confidential info / something Kai can enlighten you upon). In short, pls just merge this as it is